framework/LearningGate.py

```diff
@@ -150,10 +150,9 @@ def train(self,trainingSet):
         # the check on the number of time steps (consistency) is performed inside the historySnapShoots method
         # get the time slices
         newTrainingSet = mathUtils.historySnapShoots(trainingSet, len(self.historySteps))
-        if type(newTrainingSet).__name__ != 'list':
-          self.raiseAnError(IOError,newTrainingSet)
+        assert(type(newTrainingSet).__name__ == 'list')
         # copy the original ROM
-        originalROM = copy.deepcopy(self.supervisedContainer[0])
+        originalROM = self.supervisedContainer[0]
```

Comment:
Could you tell me the reason why the deep copy is removed?

Reply:
If you look below, each time the originalROM is used, it's surrounded in a deepcopy; therefore, we're wasting a deepcopy (which is expensive) by deepcopying here.

Reply:
ok